### DISCUSSION TITLE
does `json` option support all json values?

### DISCUSSION THREAD

**Discussion by samzilverberg** [sindresorhus/got#2008] [General]
I was trying to use GOT (latest) to write some "niche" tests that send some json that probably only people trying to break your server would use.

values such as: `true`, `false`, `null`, `"123"`, `1`, `[1,2,3]`.

But noticed that GOT only supports `Record<string, any>`.

https://github.com/sindresorhus/got/blob/d95ceeafb16f306e11f5d7251a82ba073252cbee/source/core/options.ts#L1356-L1365

Shouldn't it support all valid json values?
Is there a way to use GOT to send these "niche" json values?

side note: I'm not sure how the array is supported, I can only imagine that there is some conversion between Record and array...

**Comment by sindresorhus** [comment]
this was probably fixed at some point, the `json` option already supports all valid JSON values! the type is `unknown`, not `Record<string, any>`.

you can send primitives like this:

```js
await got.post('https://example.com', {
	json: true
});

await got.post('https://example.com', {
	json: 'hello'
});

await got.post('https://example.com', {
	json: [1, 2, 3]
});
```
